source/features/select-notifications.tsx

```diff
@@ -49,45 +49,27 @@ function getFiltersSelector(formData: FormData, category: Category): string[] {
 	return formData.getAll(category).map(value => filters[value as Filter]);
 }
 
-function handleSelection({target}: Event): void {
-	const selectAllCheckbox = $('input[type="checkbox"].js-notifications-mark-all-prompt');
-	// Reset the "Select all" checkbox
-	if (selectAllCheckbox.checked) {
-		selectAllCheckbox.click();
-	}
-
-	let excluded: HTMLElement[] = [];
-	const toggleSelectability = (elements: HTMLElement[], state: boolean): void => {
-		for (const element of elements) {
-			element.toggleAttribute('data-check-all-item', state);
-		}
-	};
-
-	if (elementExists(':checked', target as Element)) {
-		// @ts-expect-error TS bug
-		const formData = new FormData($('form#rgh-select-notifications-form'));
-		const types = getFiltersSelector(formData, 'Type');
-		const statuses = getFiltersSelector(formData, 'Status');
-		const readStatus = getFiltersSelector(formData, 'Read');
-
-		excluded = $$('.notifications-list-item')
-			.filter(notification =>
-				(types.length > 0 && !elementExists(types, notification))
-				|| (statuses.length > 0 && !elementExists(statuses, notification))
-				|| (readStatus.length > 0 && !notification.matches(readStatus)),
-			)
-			.map(notification => $('.js-notification-bulk-action-check-item', notification));
-
-		// Make excluded notifications unselectable
-		toggleSelectability(excluded, false);
-
-		// If at least one notification is selectable, trigger the "Select all" checkbox
-		if (elementExists('.js-notification-bulk-action-check-item[data-check-all-item]')) {
-			selectAllCheckbox.click();
+function handleSelection(): void {
+	// @ts-expect-error TS bug
+	const formData = new FormData($('form#rgh-select-notifications-form'));
+	const types = getFiltersSelector(formData, 'Type');
+	const statuses = getFiltersSelector(formData, 'Status');
+	const readStatus = getFiltersSelector(formData, 'Read');
+	const selectorGroups = [types, statuses, readStatus].filter(selectors => selectors.length > 0);
+	const deselectAll = selectorGroups.length === 0;
+
+	const notifications = $$('.notifications-list-item');
+	for (let index = 0; index < notifications.length; index++) {
+		const notification = notifications[index];
+		const input = $('input.js-notification-bulk-action-check-item', notification);
+
+		// Updating the "checked" property does not raise any events
+		input.checked = !deselectAll && selectorGroups.every(selectorGroup => elementExists(selectorGroup, notification));
+
+		// Trigger the selection action bar update
+		if (notifications.length - index === 1) {
+			input.dispatchEvent(new Event('change', {bubbles: true}));
 		}
```

Comment:
Why not just place `dispatchEvent` outs the loop? Then you can use a regular for-of, no `index`